src/wallet/spend.cpp

```diff
@@ -575,24 +602,17 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
     return best_result;
 }
 
-std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& available_coins, const CAmount& nTargetValue, const CCoinControl& coin_control, const CoinSelectionParams& coin_selection_params)
+std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsFund& coins, const CAmount& nTargetValue, const CCoinControl& coin_control, const CoinSelectionParams& coin_selection_params)
 {
     CAmount value_to_select = nTargetValue;
 
+    // Add the manually selected coins first
```

Comment:
In 84b4ffe85ba0267c49080e9a096e45d6af63e720 "wallet: remove fetch pre-selected-inputs responsibility from SelectCoins"

I'm wondering if we should just make this function only return a `SelectionResult` for the automatic coin selection and move the preset input stuff to the caller. If `FetchSelectedInputs` made a `SelectionResult` for the preset inputs, then the only preset input things left in `SelectCoins` would be the merging of the `SelectionResult`s, and that can be done in the caller too.

Reply:
Been thinking about this and pushed some refactoring that you might like a bit more.

The idea was based on try to not mix the Coin Selection code with the inputs fetching while still be able to remove the manual coins selection stuff from the automatic coin selection, the extra loops for the pre-set inputs and  the `FetchCoins` wrapper function.

So, have decoupled the `SelectCoins` into two functions:

1) AutomaticCoinSelection:
Which is basically the previously called `SelectCoins` without the manually selected inputs stuff.
So, it will purely receive a set of coins and be in charge of selecting the best subset of them to
cover the target amount.

2) SelectCoins:
In charge of selecting all the user manually selected coins first ("pre-set inputs"), and
if coin_control 'm_allow_other_inputs=true', call 'AutomaticCoinSelection' to select a
set of coins such that the target - pre_set_inputs.total_amount is met. Then merge
both results.